src/PersistenceProviders/Proto.Persistence.Sqlite/SqliteProvider.cs

```diff
@@ -84,14 +84,13 @@ public Task GetEventsAsync(string actorName, long indexStart, long indexEnd, Act
                     {
                         callback(JsonConvert.DeserializeObject<object>(item.EventData, new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.Auto }));
                     }
+                    return items.Any() ? items.LastOrDefault().EventIndex : -1;
```

Comment:
should looks like this instead?

return Task.FromResult(items.Any() ? items.LastOrDefault().EventIndex : -1);